src/utils/BlockUtils.ts

```diff
@@ -40,13 +49,25 @@ export async function averageBlockTime(
     return { average: cache.average, blockRange: cache.blockRange };
   }
 
-  latestBlockNumber ??= await provider.getBlockNumber();
+  if (!isDefined(latestBlockNumber)) {
```

Comment:
what's the reasoning behind this change inside this if {} branch?

Reply:
The block offset is only subtracted from `latestBlockNumber` if `latestBlockNumber` was resolved via getBlockNumber(). If the caller supplies `latestBlockNumber` then I thought it was OK to avoid applying the offset, since they're apparently explicit about wanting to know the timing between two specific blocks.

I described it briefly here, but I will update with additional context.
```
// Default offset to the latest block number. This is subtracted from the block number
// of the latest block when it is queried from the network, rather than having been
// specified by the caller.
```